### PR TITLE
Add unit tests

### PR DESCRIPTION
**Pull request by jakubjen**

### CODE REVIEW

File: src/app/app.component.spec.ts [hunk 1]
```diff
@@ -19,13 +19,17 @@ describe('AppComponent', () => {
     const app = fixture.componentInstance;
     expect(app.title).toEqual('chuck-norris-angular');
   });
+  it(`should have container'`, () => {
+    const fixture = TestBed.createComponent(AppComponent);
+    const app = fixture.componentInstance;
+    const container = fixture.nativeElement.querySelector('.container');
+    expect(container).not.toBeNull();
```
Comment (ogigg) [inline src/app/app.component.spec.ts]: `toBeTruthy` might be better idea. I think if container will be `undefined` this test will pass but I'm not 100% sure

File: src/app/shared/input/input.component.spec.ts [hunk 2]
```diff
@@ -5,18 +5,72 @@ import { InputComponent } from './input.component';
...
+    const input: HTMLInputElement =
+      fixture.nativeElement.querySelector('input');
+    const value = input.value;
+    expect(value === testData.value).toBeTrue();
+  });
+
+  it('should have input and user should be able to change the value', () => {
+    const input: HTMLInputElement =
+      fixture.nativeElement.querySelector('input');
+    const spy = spyOn(component, 'handleTyping');
```
Comment (ogigg) [inline src/app/shared/input/input.component.spec.ts]: const `handleTypingSpy`

File: src/app/shared/input/input.component.spec.ts [hunk 3]
```diff
@@ -5,18 +5,72 @@ import { InputComponent } from './input.component';
...
+    input.value = 'Super title';
+    input.dispatchEvent(new Event('input'));
+    fixture.detectChanges();
+    expect(spy).toHaveBeenCalled();
+  });
+
+  it('should have input and input should have name equal to name @input', () => {
+    const input: HTMLInputElement =
+      fixture.nativeElement.querySelector('input');
+    console.log(input);
```
Comment (ogigg) [inline src/app/shared/input/input.component.spec.ts]: console.log

File: src/app/shared/input/input.component.spec.ts [hunk 4]
```diff
@@ -5,18 +5,72 @@ import { InputComponent } from './input.component';
...
+
+  it('should have text-label element', () => {
+    const label = fixture.nativeElement.querySelector('.text-label');
+    expect(label).not.toBeNull();
+  });
+
+  it('should have text-label and text should be equal to label @input ', () => {
+    const label: HTMLElement =
+      fixture.nativeElement.querySelector('.text-label');
+    console.log(label.innerText);
```
Comment (ogigg) [inline src/app/shared/input/input.component.spec.ts]: console.log

File: src/app/shared/input/input.component.ts [hunk 5]
```diff
@@ -10,4 +10,10 @@ export class InputComponent {
   @Input() name = '';
   @Input() value = '';
   @Output() valueChange = new EventEmitter<string>();
+
+  handleTyping(event: Event): void {
+    const target = event.target as HTMLInputElement;
+    const value = target.value;
+    this.valueChange.emit(value);
+  }
```
Comment (ogigg) [inline src/app/shared/input/input.component.ts:18]: why not just 
```
handleTyping(event: Event): void {
    this.valueChange.emit(event.target.value)
  }
```
?

Comment (jakubjen) [reply src/app/shared/input/input.component.ts:18]: My mistake

File: src/app/app.component.spec.ts (hunk 1 above)
Comment (jakubjen) [reply src/app/app.component.spec.ts]: I test it and when I remove container element querySelector return null.

File: src/app/shared/input/input.component.ts (hunk 5 above)
Comment (jakubjen) [reply src/app/shared/input/input.component.ts:18]: When I use 
handleTyping(event: Event): void {
    this.valueChange.emit(event.target.value)
  }
  Ide show me objet is possibly null and error target Property 'value' does not exist on type 'EventTarget'.

**Review by ogigg — APPROVED** [review]

**Pull request merged**